lib/matplotlib/lines.py

```diff
@@ -507,7 +504,11 @@ def set_pickradius(self, d):
         d : float
             Pick radius, in points.
         """
-        self.pickradius = d
+        if not isinstance(d, Number):
+            raise ValueError("pick radius should be a distance")
```

Comment:
Great to move this.  But should it also be a positive number?

Reply:
Sure.